SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -1105,3 +1106,43 @@ void USpatialSender::UpdateInterestComponent(AActor* Actor)
 	Worker_EntityId EntityId = PackageMap->GetEntityIdFromObject(Actor);
 	Connection->SendComponentUpdate(EntityId, &Update);
 }
+
+void USpatialSender::RetireEntity(const Worker_EntityId EntityId)
+{
+	if (USpatialActorChannel* ActorChannel = NetDriver->GetActorChannelByEntityId(EntityId))
+	{
+		if (AActor* Actor = ActorChannel->GetActor())
+		{
+			if (Actor->IsNetStartupActor())
+			{
+				// In the case that this is a startup actor, we won't actually delete the entity in SpatialOS.  Instead we'll Tombstone it.
+				Receiver->RemoveActor(EntityId);
+				AddTombstoneToEntity(EntityId);
+			}
+			else
+			{
+				Connection->SendDeleteEntityRequest(EntityId);
+			}
+		}
+		else
+		{
+			UE_LOG(LogSpatialSender, Warning, TEXT("RetireEntity: Couldn't find Actor for EntityId: %lld"), EntityId);
+		}
+	}
+	else
+	{
+		UE_LOG(LogSpatialSender, Warning, TEXT("RetireEntity: Couldn't find SpatialActorChannel for EntityId: %lld"), EntityId);
+	}
+}
+
+void USpatialSender::AddTombstoneToEntity(const Worker_EntityId EntityId)
+{
+	check(NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::TOMBSTONE_COMPONENT_ID));
+	Worker_ComponentData ComponentData = Tombstone().CreateData();
+	
+	Connection->SendAddComponent(EntityId, &ComponentData);
```

Comment:
Currently components events are fedback into this worker - so this worker's view won't reflect this change. I worked around this in the dormancy work by fabricating a Worker_RemoveComponentOp, and they passing it to the connection and static component view.